### PR TITLE
feat: add linea networks support

### PR DESCRIPTION
**Pull request by VGau**
## Description

- Added linea networks support in controller utils
- Added linea networks support in network controller
- Added linea networks support in transaction controller
- Updated `@metamask/eth-json-rpc-infura` package version to 8.1.0 to support linea networks

### `@metamask/controller-utils`

- **ADDED:** Add Linea networks to `TESTNET_TICKER_SYMBOLS`, `BUILT_IN_NETWORKS`, `NETWORK_ID_TO_ETHERS_NETWORK_NAME_MAP`, `InfuraNetworkType`, `NetworkType`, `BuiltInNetworkName`, `ChainId`, `NetworkId`, and `NetworksTicker`
  - `isNetworkType` will now return true when given `linea-mainnet` or `linea-goerli`

### `@metamask/network-controller`

- **ADDED:** Add support for Infura provider types `linea-mainnet` and `linea-goerli`
- **CHANGED:** Update `@metamask/eth-json-rpc-infura` package version to 8.1.0 to support Linea networks

### `@metamask/transaction-controller`

- **CHANGED:** Update `approveTransaction`, `stopTransaction`, and `speedUpTransaction` to use proper chain ID and network ID when creating transaction for Linea networks
## Checklist

- [X] I've updated the test suite for new or updated code as appropriate
- [ ] I've updated documentation for new or updated code as appropriate (note: this will usually be JSDoc)
- [ ] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

**Review by Arschieqwe — APPROVED** [review]

**Review by Arschieqwe — APPROVED** [review]
Thanks

File: packages/transaction-controller/src/TransactionController.ts [hunk 1]
```diff
@@ -663,7 +663,11 @@ export class TransactionController extends BaseController<
       providerConfig: { type: chain, chainId, nickname: name },
     } = this.getNetworkState();
 
-    if (chain !== RPC) {
+    if (
```
Comment (mcmire) [inline packages/transaction-controller/src/TransactionController.ts:666]: I know there aren't any tests for `getCommonConfiguration`, but what do you think about adding a test for this behavior?

**Review by mcmire — COMMENTED** [review]
Looks good, but I just had one suggestion.

**Comment by mcmire** [comment]
Also, there are some differences between the "Description" and "Changes" sections in the PR description that perhaps we ought to clarify better:

- The Description is intended for you to explain the purpose and scope of your changes and share knowledge that people might not be able to see just from reading the PR. The audience for this section is other developers who are maintaining this code. They might want to know: What is motivating this change (is there a ticket or can you provide some more context)? How do the changes work? If you had to upgrade a package, why?
- The Changes is intended to be a list of your changes from the perspective of each package in the monorepo. The audience is consumers of the package (i.e. developers working on either the extension or mobile app) who will scan the changelog when they upgrade to a new version. They want to know: What are the exact changes to the API I need to be aware of (types, interfaces, functions, methods)? What are the effects on whichever platform might want to make use of these changes? 

With that in mind, something you might want to highlight in the Description is, why did you have to upgrade `@metamask/eth-json-rpc-infura`? Why did you have to exclude Linea from the `getCommonConfiguration` method?

As for Changes, I think you could borrow from what you have in the Description. What if you updated Changes to the following?

``` markdown
### `@metamask/controller-utils`

- **ADDED:** Add Linea networks to `TESTNET_TICKER_SYMBOLS`, `BUILT_IN_NETWORKS`, `NETWORK_ID_TO_ETHERS_NETWORK_NAME_MAP`, `InfuraNetworkType`, `NetworkType`, `BuiltInNetworkName`, `ChainId`, `NetworkId`, and `NetworksTicker`
  - `isNetworkType` will now return true when given `linea-mainnet` or `linea-goerli`

### `@metamask/network-controller`

- **ADDED:** Add support for Infura provider types `linea-mainnet` and `linea-goerli`
- **CHANGED:** Update `@metamask/eth-json-rpc-infura` package version to 8.1.0 to support Linea networks

### `@metamask/transaction-controller`

- **CHANGED:** Update `approveTransaction`, `stopTransaction`, and `speedUpTransaction` to use proper chain ID and network ID when creating transaction for Linea networks
```

I filled in what I thought was a proper summary of the changes to the transaction controller from what I could guess, but feel free to clarify that further if you have more knowledge.

This should help us when building the changelog for a new release.

**Comment by VGau** [comment]
> Also, there are some differences between the "Description" and "Changes" sections in the PR description that perhaps we ought to clarify better:
> 
> * The Description is intended for you to explain the purpose and scope of your changes and share knowledge that people might not be able to see just from reading the PR. The audience for this section is other developers who are maintaining this code. They might want to know: What is motivating this change (is there a ticket or can you provide some more context)? How do the changes work? If you had to upgrade a package, why?
> * The Changes is intended to be a list of your changes from the perspective of each package in the monorepo. The audience is consumers of the package (i.e. developers working on either the extension or mobile app) who will scan the changelog when they upgrade to a new version. They want to know: What are the exact changes to the API I need to be aware of (types, interfaces, functions, methods)? What are the effects on whichever platform might want to make use of these changes?
> 
> With that in mind, something you might want to highlight in the Description is, why did you have to upgrade `@metamask/eth-json-rpc-infura`? Why did you have to exclude Linea from the `getCommonConfiguration` method?
> 
> As for Changes, I think you could borrow from what you have in the Description. What if you updated Changes to the following?
> 
> ```gfm
> ### `@metamask/controller-utils`
> 
> - **ADDED:** Add Linea networks to `TESTNET_TICKER_SYMBOLS`, `BUILT_IN_NETWORKS`, `NETWORK_ID_TO_ETHERS_NETWORK_NAME_MAP`, `InfuraNetworkType`, `NetworkType`, `BuiltInNetworkName`, `ChainId`, `NetworkId`, and `NetworksTicker`
>   - `isNetworkType` will now return true when given `linea-mainnet` or `linea-goerli`
> 
> ### `@metamask/network-controller`
> 
> - **ADDED:** Add support for Infura provider types `linea-mainnet` and `linea-goerli`
> - **CHANGED:** Update `@metamask/eth-json-rpc-infura` package version to 8.1.0 to support Linea networks
> 
> ### `@metamask/transaction-controller`
> 
> - **CHANGED:** Update `approveTransaction`, `stopTransaction`, and `speedUpTransaction` to use proper chain ID and network ID when creating transaction for Linea networks
> ```
> 
> I filled in what I thought was a proper summary of the changes to the transaction controller from what I could guess, but feel free to clarify that further if you have more knowledge.
> 
> This should help us when building the changelog for a new release.

Thanks for you comment @mcmire :) I updated "Changes" part in the PR description.

Regarding this `Why did you have to exclude Linea from the getCommonConfiguration method?`: 
I needed to exclude Linea because `@ethereumjs/common` does not support Linea networks for now.

**Comment by mcmire** [comment]
@VGau Thanks! I think it would be good to add a test, even if it's just for `approveTransaction`, that guarantees that a transaction for Linea looks the way we expect, and then I can approve this PR.

**Comment by mcmire** [comment]
@VGau Is this ready for another review?

**Comment by VGau** [comment]
> @VGau Is this ready for another review?

@mcmire Yes it is ready for another review :)

**Review by mcmire — DISMISSED** [review]
This looks good to me! Sorry for the long wait on this.

**Comment by mcmire** [comment]
The test failures for `assets-controller` look intermittent — try re-running CI and they should work.

**Comment by mcmire** [comment]
@metamaskbot publish-preview

File: packages/controller-utils/src/types.ts [hunk 2]
```diff
@@ -62,12 +68,16 @@ export const NetworkId = {
...
+  [InfuraNetworkType['linea-mainnet']]: '59144',
 } as const;
 export type NetworkId = typeof NetworkId[keyof typeof NetworkId];
 
 export enum NetworksTicker {
   mainnet = 'ETH',
   goerli = 'GoerliETH',
   sepolia = 'SepoliaETH',
+  'linea-goerli' = 'LineaETH',
+  'linea-mainnet' = 'ETH',
```
[the remark this answers is not in the document]
Comment (mcmire) [reply packages/controller-utils/src/types.ts:81]: I'm verifying the effects of this.

[the remark this answers is not in the document]
Comment (danjm) [reply packages/controller-utils/src/types.ts:81]: As of this change https://github.com/MetaMask/metamask-extension/pull/14454, we started  prefixing `ETH` on all test networks with the network name.

With Linea mainnet, this is the first "main net" other than ethereum mainnet that we are supporting within our main network UIs (e.g. the network dropdown). We may want to do something different in that case.

So I would say that lego's suggestion for linea goerli is correct, but I will check with product on the change for Linea mainnet

**Comment by mcmire** [comment]
Ah, I see, this is from a fork, so we can't create a preview-build. Temp PR created here: https://github.com/MetaMask/core/pull/1478

File: packages/controller-utils/src/types.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (danjm) [reply packages/controller-utils/src/types.ts:81]: Update: product manager for the Linea project says that `'ETH'` should be the symbol for Linea network. They also specifically wanted `'LineaETH'` for linea-goerli, and that is what it currently is in the extension and and mobile.

So I'd suggest leaving these unchanged for now, and we can ask extension and mobile product managers to consider whether they want to modify these in the future.

**Comment by mcmire** [comment]
Here are the preview builds for anyone who wants to test this in extension:

```
{
  "@metamask/address-book-controller": "3.0.0-preview.be1b962",
  "@metamask/announcement-controller": "4.0.0-preview.be1b962",
  "@metamask/approval-controller": "3.4.0-preview.be1b962",
  "@metamask/assets-controllers": "9.2.0-preview.be1b962",
  "@metamask/base-controller": "3.0.0-preview.be1b962",
  "@metamask/composable-controller": "3.0.0-preview.be1b962",
  "@metamask/controller-utils": "4.1.0-preview.be1b962",
  "@metamask/ens-controller": "4.0.0-preview.be1b962",
  "@metamask/gas-fee-controller": "6.0.1-preview.be1b962",
  "@metamask/keyring-controller": "6.0.0-preview.be1b962",
  "@metamask/logging-controller": "1.0.0-preview.be1b962",
  "@metamask/message-manager": "7.0.2-preview.be1b962",
  "@metamask/network-controller": "10.3.0-preview.be1b962",
  "@metamask/notification-controller": "3.0.0-preview.be1b962",
  "@metamask/permission-controller": "4.0.1-preview.be1b962",
  "@metamask/phishing-controller": "5.0.0-preview.be1b962",
  "@metamask/preferences-controller": "4.2.0-preview.be1b962",
  "@metamask/rate-limit-controller": "3.0.0-preview.be1b962",
  "@metamask/signature-controller": "5.1.0-preview.be1b962",
  "@metamask/transaction-controller": "7.0.0-preview.be1b962"
}
```

File: packages/controller-utils/src/types.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (mcmire) [reply packages/controller-utils/src/types.ts:81]: Thanks for checking!

**Review by mcmire — APPROVED** [review]
@VGau I added some more tests yesterday so be sure to do a pull on your side if you need to add any more changes.

Regardless, everything here looks good to me.

**Pull request merged**